### PR TITLE
🧪 Add end-to-end tests for navigation

### PR DESCRIPTION
**Pull request by rye**
I like testing.

Testing is good.

"We should test more stuff" has been a long-standing thing I've said, but here I'm actually trying my hand at e2e testing.

I'm Not Sure™ if this affects coverage at all. It'd be nice if it did, but I don't think Detox instruments that yet, or if they do, I don't think we're uploading that. So, I guess I just have to give this a try.

***

list of views to test moved to #6788

### CODE REVIEW

File: e2e/streaming.spec.js [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+	it('is reachable under the home screen', async () => {
+		await expect(element(by.id('screen-homescreen'))).toBeVisible()
+		await expect(
+			element(by.id('homescreen-button-StreamingView')),
+		).toBeVisible()
+		await element(by.id('homescreen-button-StreamingView')).tap()
+		await expect(element(by.id('screen-homescreen'))).toNotBeVisible()
+		await expect(element(by.id('screen-streaming'))).toBeVisible()
+
+		await element(by.text('KSTO')).tap()
```
Comment (hawkrives) [inline e2e/streaming.spec.js]: I think there’s a way to target tab bar items explicitly

**Comment by hawkrives** [comment]
I think we should tweak the Detox tests to not reinstall the app before every test. I don’t remember why I did that originally.

**Comment by hawkrives** [comment]
Anyway, ideas:

- Calendar, tap first item in the list of there are any items

- Building Hours, same thing

- Settings, assert that the Sign In button is visible?

File: e2e/streaming.spec.js (hunk 1 above)
Comment (rye) [reply e2e/streaming.spec.js]: Ah yes, after ~10 minutes of digging: [`tabBarTestID`](https://reactnavigation.org/docs/en/material-bottom-tab-navigator.html#tabbartestid) looks like the ticket. We will set that in the `navigationOptions` under the screens in such things in addition to the e.g. `tabBarLabel` and `tabBarIcon` things.

**Comment by rye** [comment]
I have taken care of the relaunching behavior in 6dd0d1f49456d4a8272beca47a224e9b3f8fdd43.

**Comment by rye** [comment]
So I can't find my `testID` because "React Native only supports the testID prop on the native built-in components."

It sounds like I have to manually propagate that through to the appropriate element.

**Comment by rye** [comment]
Hey, a passing build!

**Comment by drewvolz** [comment]
<details>
<summary>List of views I think we could add E2E tests to. Feel free to edit.</summary>

### Home
- [ ] List of buttons

### Menus
- [ ] Stav Hall
- [ ] Cage
- [ ] Pause
- [ ] Burton
- [ ] LDC
- [ ] Weitz
- [ ] Sayles
- [ ] Menu Item Detail

### SIS
- [ ] Balances
- [ ] Course Search Home
- [ ] Course Search List
- [ ] Course Search Detail
- [ ] TES
- [ ] Open Jobs List
- [ ] Open Jobs Detail

### Building hours
- [ ] Hours List
- [ ] Hours Detail
- [ ] Hours Problem Report
- [ ] Hours Schedule Editor

### Calendar
- [ ] St. Olaf Calendar
- [ ] Oleville Calendar
- [ ] Northfield Calendar
- [ ] Calendar Event Detail

### Directory
- [ ] Link to browser (current implementation)
- [ ] Directory List (awaiting PR merge)
- [ ] Directory Contact Detail (awaiting PR merge)

### Streaming Media
- [ ] Events calendar
- [ ] Webcams
- [ ] KSTO Radio
- [ ] KRLX Radio
- [ ] Radio Schedule

### News
- [ ] St. Olaf List
- [ ] Oleville List
- [ ] Mess List
- [ ] PoliticOle List
- [ ] KSTO List

### Map
- [ ] Link to Browser

### Important Contacts
- [ ] Contacts List
- [ ] Contacts Detail

### Transit
- [ ] Express Bus
- [ ] Red Line
- [ ] Blue Line
- [ ] Oles Go
- [ ] Other Modes
- [ ] Other Modes Detail

### Dictionary
- [ ] Dictionary List
- [ ] Dictionary Detail
- [ ] Dictionary Editor

### Student Orgs
- [ ] Orgs List
- [ ] Orgs Detail

### Moodle
- [ ] Link to Browser

### Report a Problem
- [ ] List of help cards

### stoPrint
- [ ] Printer List
- [ ] Print Job Release

### Safety Concerns
- [ ] Link to Browser

### Settings
- [ ] Login
- [ ] Contact Us
- [ ] FAQs
- [ ] Reset
- [ ] Icon Settings
- [ ] Credits
- [ ] Privacy Policy
- [ ] Legal
- [ ] Contributing (link to browser)

## Smaller Fish

### Developer
- [ ] API Test
- [ ] BonApp Picker
- [ ] Redux Debug
- [ ] Sentry Message
- [ ] Sentry Exception

### Integrated but not visible
- [ ] Bus Map

</details>

**Comment by drewvolz** [comment]
I realize these have been sitting in this PR but probably can be merged in as they are written. Is there anything besides resolving conflicts that we need to do to get these in?

**Comment by hawkrives** [comment]
Riffing from @drewvolz's comment, I think that once we get Detox working again this should be only a small amount of work to merge!

**Comment by hawkrives** [comment]
``` 
  ● Streaming Media View › has the Stream List visible by default

    Test Failed: No elements found for “MATCHER(id == “homescreen-button-StreamingView”)”

    HINT: To print view hierarchy on failed actions/matches, use log-level verbose or higher.

      30 | 	it('has the Stream List visible by default', async () => {
      31 | 		// Navigate into StreamingView
    > 32 | 		await element(by.id('homescreen-button-StreamingView')).tap()
         | 		                                                        ^
      33 |
      34 | 		// The stream-list should be visible now
      35 | 		await expect(element(by.id('stream-list'))).toBeVisible()

      at Object.<anonymous> (e2e/streaming.spec.ts:32:59)
``` 

Working on this part next…

**Comment by hawkrives** [comment]
ready!

**Review by drewvolz — APPROVED** [review]

**Review by rye — COMMENTED** [review]
I can't ✅ this but I love it! This certainly seems to be going in the right direction.

As to overall strategy, I do think it's worth us exploring either mocking or full parametrization of things so that we have more control over data we expect to see in the app from the test suite and so that we can verify more tightly. However, I think we should do that in a separate MR — I think that's part of the larger infrastructurification push.

**Comment by hawkrives** [comment]
I fully agree! And I agree about that being a separate effort - we'll get a lot of value just from having the assurance that every view is reachable by navigating the app, I think!

**Pull request merged**